internal/framework/status/updater.go

```diff
@@ -213,32 +221,67 @@ func (upd *UpdaterImpl) updateGatewayAPI(ctx context.Context, statuses GatewayAP
 	}
 }
 
+// The function in wait.ExponentialBackoffWithContext will retry if it returns nil as its error,
+// which is what we want if we encounter an error from the functions we call. However,
+// the linter will complain if we return nil if an error was found.
+//
+//nolint:nilerr
 func (upd *UpdaterImpl) writeStatuses(
 	ctx context.Context,
 	nsname types.NamespacedName,
 	obj client.Object,
 	statusSetter func(client.Object),
 ) {
-	// The function handles errors by reporting them in the logs.
-	// We need to get the latest version of the resource.
-	// Otherwise, the Update status API call can fail.
-	// Note: the default client uses a cache for reads, so we're not making an unnecessary API call here.
-	// the default is configurable in the Manager options.
-	if err := upd.cfg.Client.Get(ctx, nsname, obj); err != nil {
-		if !apierrors.IsNotFound(err) {
-			upd.cfg.Logger.Error(
-				err,
-				"Failed to get the recent version the resource when updating status",
-				"namespace", nsname.Namespace,
-				"name", nsname.Name,
-				"kind", obj.GetObjectKind().GroupVersionKind().Kind)
-		}
-		return
-	}
+	err := wait.ExponentialBackoffWithContext(
+		ctx,
+		wait.Backoff{
+			Duration: time.Millisecond * 200,
+			Factor:   2,
+			Jitter:   0.5,
+			Steps:    4,
+			Cap:      time.Millisecond * 3000,
+		},
+		// Function returns true if the condition is satisfied, or an error if the loop should be aborted.
+		func(ctx context.Context) (bool, error) {
+			// The function handles errors by reporting them in the logs.
+			// We need to get the latest version of the resource.
+			// Otherwise, the Update status API call can fail.
+			// Note: the default client uses a cache for reads, so we're not making an unnecessary API call here.
+			// the default is configurable in the Manager options.
+			if err := upd.cfg.Client.Get(ctx, nsname, obj); err != nil {
+				// apierrors.IsNotFound(err) can happen when the resource is deleted,
+				// so no need to retry or return an error.
+				if apierrors.IsNotFound(err) {
+					upd.cfg.Logger.V(1).Info(
+						"Resource was not found when trying to update status",
+						"namespace", nsname.Namespace,
+						"name", nsname.Name,
+						"kind", obj.GetObjectKind().GroupVersionKind().Kind)
+					return true, nil
+				}
+				upd.cfg.Logger.V(1).Info(
+					"Encountered error when getting resource to update status",
```

Comment:
Let's add the error to the log message as a key value pair. 

I also think it's worth adding to the log message that we are going to retry

Reply:
I was thinking about adding something mentioning "retry" to the log message, but I was struggling with determining if I should because I don't think there's a way to tell if we are on the last retry. However, does that even matter if we are on the last retry but then log that we are retrying again but we don't?

Reply:
Good point, I didn't think of that. Let's just leave retrying out.